feat(stark-ui): implement OnEnterKey directive and module

christophercr opened the pull request:
ISSUES CLOSED: #538 #548

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #538 #548 


## What is the new behavior?
OnEnterKey directive and module implemented

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

christophercr commented:
PR updated with your remarks. Can you please review it?

SuperITMan commented:
I'm reviewing it. It should be ok in 15-20 minutes

SuperITMan commented:
Could you please provide an example for the usage of this directive ?
I have some troubles to make it work. 

The code which is working is this one:
```html
<input name="test" [(ngModel)]="valueTest" [starkOnEnterKey]="onEnterKey" [starkOnEnterKeyParams]="[valueTest]" type="text">
```

```typescript
public valueTest: any;
```

If I put this code, it does not work.
```html
<input name="test" [starkOnEnterKey]="onEnterKey" [starkOnEnterKeyParams]="[valueTest]" type="text">
```

But if we need to set ngModel, I don't get the idea of starkOnEnterKeyParams 😕

christophercr commented:
@SuperITMan in your example that does not work: what is "valueTest" supposed to be?

It should be a property defined in your component. If you want to pass a value directly, then you need to pass it with quotation marks so Angular can pass it as string:

```html
<input name="test" [starkOnEnterKey]="onEnterKey" [starkOnEnterKeyParams]="['valueTest']" type="text"> 
```

catlabs commented:
I created a ticket to add a page in the showcase #548

christophercr commented:
@SuperITMan I updated this PR by adding an example in the Showcase.

I also renamed the module for this directive to "KeyboardDirectivesModule" as we discussed so it will also contain other keyboard related directives such as RestrictInput.

SuperITMan commented on showcase/src/app/demo/keyboard-directives/keyboard-directives.component.ts:
```diff
@@ -0,0 +1,48 @@
+
+	public onEnterKeyCallback(...paramValues: any[]): void {
+		if (typeof this.latestInputValue !== "undefined") {
+			this.latestInputValue = paramValues[1];
```
I think you do a method 'onEnterKeyCallbackValueX' for each input example.
This method could obviously call a common method like `onEnterKeyCallback(latestInputValue, paramValues)`

And so for the latestInputValue use `this.inputValueX` instead of paramValues[1] which is actually not the latestInputValue the component got. 
We should be sure that ngModel is working....

SuperITMan replied:
And latestInputValue could be changed by ngModelLatestValue or something like this 😊

christophercr commented:
@SuperITMan I've updated my PR with the changes we discussed. Can you have a look and merge if everything is Ok?
